ignite/contrib/metrics/regression/geometric_mean_absolute_error.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import division
+import torch
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric
+import numpy as np
+
+
+class GeometricMeanAbsoluteError(Metric):
+    r"""
+    Calculates the Geometric Mean Absolute Error.
+    It has been proposed in `Botchkarev 2018`__.
+    :math:`\text{GMRAE} = \sqrt[n]{\prod_{j=1}^{\n}{|A_j - P_j|}`
+    Where, :math:`A_j` is the ground truth and :math:`P_j` is the predicted value.
+    - `update` must receive output of the form `(y_pred, y)`.
+    - `y` and `y_pred` must be of same shape.
+    __ https://arxiv.org/abs/1809.03006
+    """
+
+    def reset(self):
+        self._product_of_errors = 1.0
+        self._num_examples = 0
+
+    def update(self, output):
+        y_pred, y = output
+        errors = torch.abs(y.view_as(y_pred) - y_pred)
+        self._product_of_errors = self._product_of_errors * torch.prod(errors).item()
+        self._num_examples += y.shape[0]
+
+    def compute(self):
+        if self._num_examples == 0:
+            raise NotComputableError('GeometricMeanAbsoluteError must have at'
+                                     'least one example before it can be computed.')
+        return np.power(self._product_of_errors, 1.0 / self._num_examples)
```

Comment:
I suggest to replace `np.power(a, b)` with `a ** b`, since numpy is not a dependency of ignite.

Reply:
Agree will incorporate tonight.